assets/app/scripts/services/apiService.js

```diff
@@ -0,0 +1,268 @@
+'use strict';
+/* jshint unused:false */
+
+// see gist: https://gist.github.com/benjaminapetersen/540ac605d3c1b0660062
+// for discussion leading up to the creation of this file
+
+angular.module('openshiftConsole')
+  .factory('APIService', function($window, API_CFG, APIGROUP_CFG) {
+
+    var defaultVersionForGroup = {
+      'extensions':   'v1beta1',
+      'experimental': 'v1beta1',
+      '':             'v1'  // this group applies to legacy openshift & k8s (/oapi & /api)
+    };
+
+
+    // {group: '', resource: '', version: ''}
+    var findAPIFor = function(qualified) {
+      var resource = qualified.resource;
+      // handle API Groups
+      if(qualified.group && qualified.group.length) {
+        // TODO: cannot validate until we have API discovery :(
+        return {
+          hostPort: APIGROUP_CFG.hostPort,
+          prefix: APIGROUP_CFG.prefix,
+          version: qualified.version || defaultVersionForGroup[qualified.group]
+        };
+      }
+
+      // else, legacy
+      var found = _.find(API_CFG, function(api) {
+        return api.resources[_.first(resource.split('/'))] || api.resources['*']; // ['*'] ?
+      });
+
+      return found ?
+              {
+                hostPort: found.hostPort,
+                prefix: found.prefixes[qualified.version] ||
+                        found.prefixes[defaultVersionForGroup[qualified.group]] ||
+                        found.prefixes['*'],
+                version: qualified.version || defaultVersionForGroup[qualified.group]
```

Comment:
and compute using '' as the group name (we know we're in the empty legacy group now), not qualified.group